src/data-types/mailstream_ssl.c

```diff
@@ -361,7 +361,7 @@ static int wait_SSL_connect(int s, int want_read, time_t timeout_seconds)
     return -1;
   }
 
-  if (pfd.revents & pfd.events) {
+  if (pfd.revents & pfd.events != pfd.events) {
```

Comment:
Current condition "pfd.revents & pfd.events" is equal "FD_ISSET(s, &fds)", but required "!FD_ISSET(s, &fds)". It is equal "pfd.revents & pfd.events != pfd.events".